### PR TITLE
Performance: limit Sentry tracking, fix NFT hooks

### PR DESCRIPTION
**Pull request by christianbaroni**
Fixes APP-1555
Fixes APP-1566

## What changed (plus any additional context for devs)
- Limits Sentry tracking to crash reporting, startup time tracking, and explicit error reporting
  - Not fully clear why but Sentry is badly hurting app performance — limiting it to tracking crashes and startup times solves the problem
- Fixes issues causing NFT data to be excessively recreated — some more improvements to be made but the big problems are resolved
  - Had to bump the `nftsQueryKey` `persisterVersion` because the structure of `data` (from the `useQuery` hook) is now different with the NFT data transformation moved into the `useQuery`.
  - The structure of the data returned by the `useLegacyNFTs` wrapper hook is unchanged and fully backwards compatible — ideally we later remove the wrapper hook and migrate to the new NFT types.
- Removes non-critical ENS-related hooks that were making a significant number of network requests
- Removes unused code in `App.js` and in the `WalletScreen` component

Also:
- Isolates the `dapps` data from UI components in the browser, because the `dapps` object is quite heavy and it's unnecessary for it to cause re-renders in search

### CODE REVIEW

File: src/utils/logger.ts [hunk 1]
```diff
@@ -72,8 +72,7 @@ const Logger = {
       console.log(...args); // eslint-disable-line no-console
     }
     if (args.length === 1 && typeof args[0] === 'string') {
-      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
-      sentryUtils.addInfoBreadcrumb.apply(null, args);
+      sentryUtils.addInfoBreadcrumb.apply(null, [args[0]]);
```
Comment (walmat) [inline src/utils/logger.ts:75]: nice

**Comment by jinchung** [comment]
I'm still reviewing - don't merge yet pls 🙏

**Review by jinchung — COMMENTED** [review]
Comments inline

File: src/resources/nfts/index.ts [hunk 2]
```diff
@@ -31,48 +31,83 @@ export function useNFTs(): NFT[] {
...
-      ),
-    [nfts]
-  );
-
   return {
-    data: { nfts, nftsMap },
+    data: (data ?? FALLBACK_DATA) as UseLegacyNFTsReturnType<TSelected>,
     error,
-    isInitialLoading: !data?.length && isFetching,
+    isInitialLoading: !data && isFetching,
```
[the remark this answers is not in the document]
Comment (christianbaroni) [reply src/resources/nfts/index.ts:98]: `data` (from the `useQuery` hook) is undefined in the initial fetch case - the fallback is only applied after `fetchNFTData` runs, so on first fetch or if there's nothing in the cache then `data` will be undefined, which I believe is what `isInitialLoading` is meant to identify.

Lots of further improvements to make to this code. The main goal to start was to ensure we're leveraging `useQuery` to share the NFT data references and avoid redoing all of this processing for every instance of `useCollectible` (1x per NFT) or `useLegacyNFTs`.

And also to do it without requiring changes outside of those two hooks, so keeping the structure of the `data` that's returned by `useLegacyNFTs` exactly the same. I think any changes to the data structure or the unique IDs has greater potential to break things, and want to keep these changes minimal and quick.

File: src/resources/nfts/index.ts [hunk 3]
```diff
@@ -31,48 +31,83 @@ export function useNFTs(): NFT[] {
...
+    ...config,
   });
 
-  const nfts = useMemo(() => data ?? [], [data]);
-
-  const nftsMap = useMemo(
-    () =>
-      nfts.reduce(
-        (acc, nft) => {
-          // index by both uniqueId and fullUniqueId bc why not
```
[the remark this answers is not in the document]
Comment (christianbaroni) [reply src/resources/nfts/index.ts:62]: Left a todo, I don't think there's any justifiable reason for it. The `nftsMap` is only used in a few places, we should be able to pretty easily delete it and consolidate to one unique ID format.

The benefits of the shared `useQuery` data processing/references are much greater than this optimization so leaving this for followup work.

File: src/screens/WalletScreen/index.tsx [hunk 4]
```diff
@@ -94,14 +86,6 @@ const WalletScreen: React.FC<any> = ({ navigation, route }) => {
     }
   }, [dangerouslyGetState, removeFirst]);
 
-  const { isEmpty: isAccountEmpty } = useAccountEmptyState(isSectionsEmpty, isLoadingUserAssets);
```
[the remark this answers is not in the document]
Comment (christianbaroni) [reply src/screens/WalletScreen/index.tsx:97]: Deleted the hook, left the methods for now — the empty state is being set in a few places. Doesn't look like it's read but would want to double check. Hope to delete that whole `accountLocal` file soon.

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**